internal/pkg/cache/redis/redis.go

```diff
@@ -33,23 +43,40 @@ type Redis struct {
 func NewRedis(address string) *Redis {
 	cfg := &config.RedisConfig{
 		Address: address,
+		RedisMode: RedisModeSingle,
 	}
 
 	return NewRedisWithConfig(cfg)
 }
 
 // NewRedisWithConfig creates a new Redis instance with configurable options.
 func NewRedisWithConfig(cfg *config.RedisConfig) *Redis {
-	o := &redis.UniversalOptions{
-		Addrs:           parseRedisAddress(cfg.Address),
-		MinIdleConns:    cfg.MinIdleConns,
-		PoolSize:        cfg.PoolSize,
-		ConnMaxIdleTime: cfg.IdleTimeout,
-		ConnMaxLifetime: cfg.MaxConnAge,
+	var c redis.UniversalClient
+
+	if cfg.RedisMode == RedisModeCluster {
+		o := &redis.ClusterOptions{
+			Addrs:           parseRedisClusterMultiAddress(cfg.Address),
```

Comment:
How is this working with the Redis Cluster discovery endpoint? Is the discovery endpoint a list of IPs, or is it always one IP?

Reply:
https://cloud.google.com/memorystore/docs/cluster/cluster-node-specification#discovery_endpoint

TL;DR: The discovery endpoint guarantees a consistent IP for the entire lifecycle of the Redis Cluster, independently of the nodes going up or down. Once the client connects to the endpoint, the client starts receiving packets with the master and replica nodes it must create a connection with.